CICE implementation

SmithJos13 opened the pull request:
<!-- INSTRUCTIONS: 
- PLEASE READ/FOLLOW THE DIRECTIONS IN EACH SECTION
- Complete the 'Commit Queue Requirements' below
- Please use github markup as much as possible (https://docs.github.com/en/get-started/writing-on-github)
- Please leave your PR in a draft state until all underlying work is completed.
-->
## Commit Queue Requirements:
<!--
- Please complete the items that follow this.
- Please "check off" completed items. Use [X] for a filled in checkbox or leave it [ ] for an empty checkbox
- Your PR will not be considered until all requirements are met.
- THIS IS YOUR RESPONSIBILITY
 -->
- [ ] Fill out all sections of this template.
- [ ] All sub component pull requests have been reviewed by their code managers.
- [ ] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] Commit 'test_changes.list' from previous step
---
## Description:
<!--
Please provide a detailed verbose description of what this PR does
-->

This PR includes all the code to bring CICE forward as a model component. It couples CICE to CMEPS mediator and is set up to run in standalone and coupled configurations. In coupled configurations, schism will interface with schism and the relevant variables have been added to SCHISM to handle CICE inputs. 

Other quality of life updates.

Code for Ice restoring time-varying boundary conditions along northern edge of CICE domain. This code can be expanded to other edges of the domain in the future.

CICE model subcyling. In its default configuration (ufs-weather model) CICE must be coupled at every single time step which can cause slow-downs. There has been code added to subcyle the model when not coupling at every single time step.

### Commit Message:
<!--
Please provide concise information for The UFS-WM and/or each sub-component:
Please delete what is not needed.
-->
```
* UFSWM - 
  * CICE - ESMF coupling vars, open boundary conditions 
  * CMEPS - CICE coupling vars added
  * SCHISM - CICE coupling vars added
  * SCHISM-ESMF -CICE coupling vars added
```

### Priority:
<!--
Please provide the priority you would prefer this pull request to have.
* Normal.
Please delete the ones that are not applicable
-->
* Normal

## Git Tracking
### UFSWM:
* CICE: https://github.com/oceanmodeling/CICE/pull/2
   * Icepack: https://github.com/oceanmodeling/Icepack/pull/3
* CDEPS: https://github.com/oceanmodeling/CDEPS/pull/3
* CMEPS: https://github.com/oceanmodeling/CMEPS/pull/4
* SCHISM: https://github.com/oceanmodeling/schism/pull/12
* SCHISM-ESMF: https://github.com/oceanmodeling/schism-esmf/pull/8

---
## Changes
### Regression Test Changes (Please commit test_changes.list):
<!--
Please let us know if this PR creates new baselines, changes baselines or not.
Please delete what is not needed.
Please make sure you have properly submitted test_changes.list
-->
* PR Adds New Tests/Baselines.
* PR Updates/Changes Baselines.
* No Baseline Changes.


<!-- STOP!!! THE FOLLOWING IS FOR CODE MANAGERS ONLY. PLEASE DO NOT FILL OUT -->
## Testing Log:
- RDHPCS
  - [ ] Hera
  - [ ] Orion
  - [ ] Hercules
  - [ ] Jet
  - [ ] GaeaC5
  - [ ] GaeaC6
  - [ ] Derecho
  - [ ] Frontera
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

SmithJos13 committed:
updated cmeps pointer

SmithJos13 committed:
updated CMEPS pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
updated icepack

SmithJos13 committed:
updated icepack

SmithJos13 committed:
updated CICE

SmithJos13 committed:
updated schism pointes

SmithJos13 committed:
updated SCHISM-ESMF pointer

SmithJos13 committed:
updated icepack pointer

SmithJos13 committed:
updated CICE/icepack pointers

SmithJos13 committed:
updated icepack/CICE pointer

SmithJos13 committed:
updated pointers

SmithJos13 committed:
Merge branch 'feature/cice' of https://github.com/SmithJos13/ufs-weather-model into feature/cice

SmithJos13 committed:
updated pointers

SmithJos13 committed:
updated pointers

SmithJos13 committed:
updated pointers

SmithJos13 committed:
updated pointers

SmithJos13 commented:
@uturuncoglu Okay, I think I fixed everything and the model compiles! Ready for a check over from you.

uturuncoglu commented:
@SmithJos13 Okay. thanks for all your effort. Let me try to move your CICE branch to UFS WM and run one of the CICE RT to see what happens. I'll update you about it.

SmithJos13 committed:
updated field dictionary

uturuncoglu commented:
@SmithJos13 I think your `fd_ufs.yaml` is not complete. I am getting error from `So_hmix`. It seems that you introduce it in the CICE cap but I am not seeing any entry in the dictionary. Could you double check?

SmithJos13 committed:
updated fd_yaml.ufs

SmithJos13 commented:
@uturuncoglu Okay, I added the variable. Hopefully that will fix it!

uturuncoglu commented:
@SmithJos13 As I suspected before, the UFS WM RT (cpld_control_p8) is failing with following error,

```
20250707 172813.133 ERROR            PET172 (ice_comp_nuopc):(ModelAdvance)  CICE clock not in sync with ESMF model clock
20250707 172813.133 ERROR            PET172 ICE:src/addon/NUOPC/src/NUOPC_ModelBase.F90:2218 Failure  - Passing error in return code
20250707 172813.133 ERROR            PET172 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:3725 Failure  - Phase 'RunPhase1' Run for modelComp 5 did not return ESMF_SUCCESS
20250707 172813.133 ERROR            PET172 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:3963 Failure  - Passing error in return code
20250707 172813.133 ERROR            PET172 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:3639 Failure  - Passing error in return code
20250707 172813.133 ERROR            PET172 UFS.F90:414 Failure  - Aborting UFS
```

I think this is related with the change that you did in the CICE cap. We need to revert that change and solve the problem in your CICE configuration with different way. Could you share your run directory with me. So, I could check your configıration.

SmithJos13 commented:
Well that is unfortunate to hear, which directory do you want access to? my Shinnecock domain? or the Bering Sea model?

uturuncoglu commented:
@SmithJos13 Yes but this is a part of process. We need to solve this before merging PR. Anyway, it would be better to use the realistic case at this point.

SmithJos13 commented:
Okay I'll send you my working directory for the Bering Sea that we can work on. 

the link is

```
/work2/noaa/vdatum/jsmith/dev/SCHISM_CICE_BER 
```

I have given you access to ufs-configure. I'll give you access to the other files in the directory as well. Let me know if there is anything particular that you need access to let know.

I can't get the model to run at the moment I think something might have been chnaged with spack-stack? I might need to rebuild the run directory, what do you think?

```
 0: Abort with message No such file or directory in file /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/cache/build_stage/spack-stage-parallelio-2.5.10-rdwrsedxim2wqcpndlxgk7wdzc3cdtra/spack-src/src/clib/pioc_support.c at line 2832
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
   1: Abort with message No such file or directory in file /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/cache/build_stage/spack-stage-parallelio-2.5.10-rdwrsedxim2wqcpndlxgk7wdzc3cdtra/spack-src/src/clib/pioc_support.c at line 2832
  80: Abort with message No such file or directory in file /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/cache/build_stage/spack-stage-parallelio-2.5.10-rdwrsedxim2wqcpndlxgk7wdzc3cdtra/spack-src/src/clib/pioc_support.c at line 2832
 240: Abort with message No such file or directory in file /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/cache/build_stage/spack-stage-parallelio-2.5.10-rdwrsedxim2wqcpndlxgk7wdzc3cdtra/spack-src/src/clib/pioc_support.c at line 2832
 320: Abort with message No such file or directory in file /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/cache/build_stage/spack-stage-parallelio-2.5.10-rdwrsedxim2wqcpndlxgk7wdzc3cdtra/spack-src/src/clib/pioc_support.c at line 2832
 160: Abort with message No such file or directory in file /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/cache/build_stage/spack-stage-parallelio-2.5.10-rdwrsedxim2wqcpndlxgk7wdzc3cdtra/spack-src/src/clib/pioc_support.c at line 2832
   2: Abort with message No such file or directory in file /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/cache/build_stage/spack-stage-parallelio-2.5.10-rdwrsedxim2wqcpndlxgk7wdzc3cdtra/spack-src/src/clib/pioc_support.c at line 2832
  81: Abort with message No such file or directory in file /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.6.0/cache/build_stage/spack-stage-parallelio-2.5.10-rdwrsedxim2wqcpndlxgk7wdzc3cdtra/spack-src/src/clib/pioc_support.c at line 2832
```

SmithJos13 commented:
Hi Ufuk,

I was doing some work this morning on the PR testing. I reran the test with my changes in the CICE cap (the subcylcing the model). I just wanted to see what was causing the issue that you were running into previously. I couldn't reproduce the error where the model wasn't syncing. Maybe there is a piece of code wasn't pushed to PR and that is why you were having issues.  

With the commit hash [4658f15](https://github.com/SmithJos13/CICE/commit/4658f15c561d401023df4691dd67d0617ece45a6), which is my most recent push to my CICE repo.

The test this time didn't pass because:

```
baseline dir = /work/noaa/epic/hercules/UFS-WM_RT/NEMSfv3gfs/develop-20250507/cpld_control_p8_intel
working dir  = /work2/noaa/vdatum/jsmith/jsmith/stmp/jsmith/FV3_RT/rt_1884995/cpld_control_p8_intel
Checking test cpld_control_p8_intel results ....
 Comparing sfcf021.tile1.nc .....USING NCCMP......NOT IDENTICAL
 Comparing sfcf021.tile2.nc .....USING NCCMP......NOT IDENTICAL

\..
...
...
../

 Comparing 20210323.060000.out_pnt.ww3 .....USING CMP......NOT IDENTICAL
 Comparing 20210323.060000.out_grd.ww3 .....USING CMP......NOT IDENTICA\

  0: The total amount of wall time                        = 594.091732
  0: The maximum resident set size (KB)                   = 2204452

Test cpld_control_p8_intel FAIL

```

I have removed some stuff from the log, but it was failing because of files not being identical on the output.

Looking at the log files it seems like CICE was able to complete the integration:

```
(ice_comp_nuopc) -------------------------------------------------------------------------
CICE: end of main integration loop
(ice_comp_nuopc) -------------------------------------------------------------------------
```

and the out file in the run directory says that it was able to complete:

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 594.091732
  0: The total amount of time in user mode                = 502.732364
  0: The total amount of time in sys mode                 = 8.664848
  0: The maximum resident set size (KB)                   = 2204452
  0: Number of page faults without I/O activity           = 777714
  0: Number of page faults with I/O activity              = 8753
  0: Number of times filesystem performed INPUT           = 31296
  0: Number of times filesystem performed OUTPUT          = 2011272
  0: Number of Voluntary Context Switches                 = 37440
  0: Number of InVoluntary Context Switches               = 4891
  0: *****************END OF RESOURCE STATISTICS*************************
  0:
Model ended:     Mon Jul 21 12:11:01 CDT 2025
```

So I think we don't need to worry about investigating alternative coupling strategies at the moment. I'm also open to hear further oppions you may have. I think those changes will be okay because they are not causing the model syncing issue. So we can run a head and just keep working on finding out why the model is failing now?

Other notes:

### **Performance:**
In my dummy domain I have run the model with the following configuration (7-days of model time Shinnecock domain):

**PET layout:**

For the intial test I have layed PET bounds as:

```
MED: 0-100
ICE: 3-99
ATM: 0-2
OCN: 100-239
```

**Non-subscyling solution (NSCS):**
```
runSeq::
@3600
  MED med_phases_prep_atm
  MED -> ATM :remapMethod=redist
  MED med_phases_prep_ocn_accum
  MED med_phases_prep_ocn_avg
  MED -> OCN :remapMethod=redist
  ATM
  OCN
  @100
    MED med_phases_prep_ice
    MED -> ICE :remapMethod=redist
    ICE
    ICE -> MED :remapMethod=redist
    MED med_phases_post_ice
  @
  ATM -> MED :remapMethod=redist
  MED med_phases_post_atm
  OCN -> MED :remapMethod=redist
  MED med_phases_post_ocn
  MED med_phases_history_write
  MED med_phases_restart_write
@
::
```

**NSCS MED: 0-100 run**

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 124.483828
  0: The total amount of time in user mode                = 116.168644
  0: The total amount of time in sys mode                 = 3.040860
  0: The maximum resident set size (KB)                   = 344768
  0: Number of page faults without I/O activity           = 76572
  0: Number of page faults with I/O activity              = 5571
  0: Number of times filesystem performed INPUT           = 123584
  0: Number of times filesystem performed OUTPUT          = 1248
  0: Number of Voluntary Context Switches                 = 10109
  0: Number of InVoluntary Context Switches               = 1057
  0: *****************END OF RESOURCE STATISTICS*************************
  0:
```



**subscyling solution:(SCS)**

```
runSeq::
@3600
  MED med_phases_prep_atm
  MED -> ATM :remapMethod=redist
  MED med_phases_prep_ocn_accum
  MED med_phases_prep_ocn_avg
  MED -> OCN :remapMethod=redist
  MED med_phases_prep_ice
  MED -> ICE :remapMethod=redist 
  ATM
  OCN
  ICE
  ATM -> MED :remapMethod=redist
  MED med_phases_post_atm
  OCN -> MED :remapMethod=redist
  MED med_phases_post_ocn
  ICE -> MED :remapMethod=redist
  MED med_phases_post_ice      
  MED med_phases_history_write
  MED med_phases_restart_write
@
::
```

**SCS MED 0-100 run**

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 62.419716
  0: The total amount of time in user mode                = 62.579111
  0: The total amount of time in sys mode                 = 2.273994
  0: The maximum resident set size (KB)                   = 344404
  0: Number of page faults without I/O activity           = 70420
  0: Number of page faults with I/O activity              = 4626
  0: Number of times filesystem performed INPUT           = 8
  0: Number of times filesystem performed OUTPUT          = 1192
  0: Number of Voluntary Context Switches                 = 6798
  0: Number of InVoluntary Context Switches               = 2473
  0: *****************END OF RESOURCE STATISTICS*************************

```

**Summary:**

So we have: 62.419716 seconds (subcycling) vs 124.483828 seconds (nonsubcycling) which means that there is a factor of 2 speed up!!!!! 


### Further tests:

I wasn't satisfied with the results, so I kept messing with the config: 

This is configuration 1 (the non-subcycling):

**NSCS MED: 0-239 100% of access**
Giving the mediator access to more PETS yields the follow result (MED: 0-239, same configuration for nonsubcycling)

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 117.467686
  0: The total amount of time in user mode                = 117.064623
  0: The total amount of time in sys mode                 = 2.755088
  0: The maximum resident set size (KB)                   = 313300
  0: Number of page faults without I/O activity           = 67167
  0: Number of page faults with I/O activity              = 5486
  0: Number of times filesystem performed INPUT           = 16
  0: Number of times filesystem performed OUTPUT          = 1232
  0: Number of Voluntary Context Switches                 = 6719
  0: Number of InVoluntary Context Switches               = 1423
  0: *****************END OF RESOURCE STATISTICS*************************
  0:
```

Slightly improved performance but essentially the same.


**NSCS MED: 0-99 ~41% access**

Giving the ocean its own dedicated PETS yields improved preformance but still a slower result:

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 81.489374
  0: The total amount of time in user mode                = 73.866153
  0: The total amount of time in sys mode                 = 2.852678
  0: The maximum resident set size (KB)                   = 311292
  0: Number of page faults without I/O activity           = 61222
  0: Number of page faults with I/O activity              = 4859
  0: Number of times filesystem performed INPUT           = 120888
  0: Number of times filesystem performed OUTPUT          = 1224
  0: Number of Voluntary Context Switches                 = 8169
  0: Number of InVoluntary Context Switches               = 720
  0: *****************END OF RESOURCE STATISTICS*************************
  0:
```

**NSCS MED: 0-72 ~30% of access**

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 86.473320
  0: The total amount of time in user mode                = 79.028150
  0: The total amount of time in sys mode                 = 2.755867
  0: The maximum resident set size (KB)                   = 311724
  0: Number of page faults without I/O activity           = 69173
  0: Number of page faults with I/O activity              = 5021
  0: Number of times filesystem performed INPUT           = 120864
  0: Number of times filesystem performed OUTPUT          = 1256
  0: Number of Voluntary Context Switches                 = 9177
  0: Number of InVoluntary Context Switches               = 1020
  0: *****************END OF RESOURCE STATISTICS*************************
```


**NSCS MED: 0-72 ~20% of access**

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 83.646298
  0: The total amount of time in user mode                = 75.072084
  0: The total amount of time in sys mode                 = 2.601845
  0: The maximum resident set size (KB)                   = 314376
  0: Number of page faults without I/O activity           = 69812
  0: Number of page faults with I/O activity              = 5138
  0: Number of times filesystem performed INPUT           = 121040
  0: Number of times filesystem performed OUTPUT          = 1224
  0: Number of Voluntary Context Switches                 = 7603
  0: Number of InVoluntary Context Switches               = 787
  0: *****************END OF RESOURCE STATISTICS*************************
```

**NSCS MED: 0-24 ~10% of access**

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 87.228606
  0: The total amount of time in user mode                = 79.934632
  0: The total amount of time in sys mode                 = 2.429447
  0: The maximum resident set size (KB)                   = 319936
  0: Number of page faults without I/O activity           = 61330
  0: Number of page faults with I/O activity              = 5134
  0: Number of times filesystem performed INPUT           = 120904
  0: Number of times filesystem performed OUTPUT          = 1216
  0: Number of Voluntary Context Switches                 = 8835
  0: Number of InVoluntary Context Switches               = 682
  0: *****************END OF RESOURCE STATISTICS*************************
```

**NSCS MED: 0-12 ~5% of access**

Not enough resources to be running optimally....

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 139.844849
  0: The total amount of time in user mode                = 130.255288
  0: The total amount of time in sys mode                 = 2.672441
  0: The maximum resident set size (KB)                   = 331036
  0: Number of page faults without I/O activity           = 76543
  0: Number of page faults with I/O activity              = 4689
  0: Number of times filesystem performed INPUT           = 120864
  0: Number of times filesystem performed OUTPUT          = 1288
  0: Number of Voluntary Context Switches                 = 8741
  0: Number of InVoluntary Context Switches               = 2019
  0: *****************END OF RESOURCE STATISTICS*************************
```

**SCS MED 0-239 100% of access**

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 66.586808
  0: The total amount of time in user mode                = 66.602336
  0: The total amount of time in sys mode                 = 2.384620
  0: The maximum resident set size (KB)                   = 316136
  0: Number of page faults without I/O activity           = 64828
  0: Number of page faults with I/O activity              = 6075
  0: Number of times filesystem performed INPUT           = 32
  0: Number of times filesystem performed OUTPUT          = 1216
  0: Number of Voluntary Context Switches                 = 9494
  0: Number of InVoluntary Context Switches               = 615
  0: *****************END OF RESOURCE STATISTICS*************************
```

**SCS MED 0-24 ~10% of access**

```
  0: *****************RESOURCE STATISTICS*******************************
  0: The total amount of wall time                        = 61.910966
  0: The total amount of time in user mode                = 55.051922
  0: The total amount of time in sys mode                 = 2.514480
  0: The maximum resident set size (KB)                   = 311436
  0: Number of page faults without I/O activity           = 63766
  0: Number of page faults with I/O activity              = 4598
  0: Number of times filesystem performed INPUT           = 120880
  0: Number of times filesystem performed OUTPUT          = 1200
  0: Number of Voluntary Context Switches                 = 10020
  0: Number of InVoluntary Context Switches               = 521
  0: *****************END OF RESOURCE STATISTICS*************************
```


Summary of bellow where % change is accessed from **SCS MED: 0-100**

| Config    | Computation time | % change from SC |
| -------- | ------- | ------- |
| SCS MED: 0-239     |   66.586808  | +6%|
| **SCS MED: 0-100**     |   62.419716  | +0%|
| SCS MED: 0-24       |   61.910966  | -1%|
| NSCS MED: 0-239  | 117.467686    | +88% |
| NSCS MED: 0-100  | 124.483828     | +100%|
| NSCS MED: 0-99    | 81.489374     |  +30%|
| NSCS MED: 0-72    | 86.473320     |  +38%|
| NSCS MED: 0-48    | 83.646298     | +33%|
| NSCS MED: 0-24    | 87.228606     | + 40%|
| NSCS MED: 0-12    | 139.844849   | + 224%| 

seem like things are plateauing around +30%

uturuncoglu commented:
@SmithJos13 Let me check with my version of UFS WM. Since UFS Coastal not synced with the UFS WM yet. It uses different baseline.

SmithJos13 commented:
@uturuncoglu okay let me know what you find out!

uturuncoglu commented:
@SmithJos13 Did you push your changes reverting your changes in CICE cap? I am still seeing changes in the model level. Maybe you pushed into your CICE branch but not updated the model? Could you double check?

SmithJos13 commented:
@uturuncoglu I didn't revert my changes and push the results because on my side the model didn't run into the same error when I ran the RT. I didn't run into any issues besides the model failing. 

I think in the future we should really investigate the sub-cycling issue. My sensitivity tests above show that regardless of load balancing, the model performance still suffers (40% slower performance). I think that these issues are only going to get worse when moving to the Bering Sea. If my sensitivity tests can be extrapolated, then we can expect 20days/8hour without the changes vs 30days/8hours with the changes. I think we should find a way to bring this functionality in the future because I believe that it is important, but in the interest of moving things along we can shelf it for now. 

I'll remove the changes so that we can move forward on this and let you know when I push the changes.

uturuncoglu commented:
@SmithJos13 Thanks for update and for your effort. Do not worry about the performance at this point. If your case is running without changes in the cap (except additional variables) then that is fine and we could move forward. We could tackle with slowness later. Maybe we could play with run sequence or balance the component. Will see. Once we merge this we could run the RT with different options to see what happens. If we really need some modification in the cap, we need to bring all parties to a dedicated call and discuss since that will also effect the others. Anyway, Let me know when you push the changes and I could test again with UFS WM. If you have any additional question etc, we could have call next week.

SmithJos13 committed:
updated CICE pointer

SmithJos13 commented:
@uturuncoglu Okay, they are reverted and the model compiles! Let me know if you have any further issues and I'll work to fix it!

SmithJos13 commented:
Where are the baselines for UFS-coastal located? I can see I can get the model to pass the tests on my end with the proper baselines. @uturuncoglu

uturuncoglu commented:
@SmithJos13 They are under `/work2/noaa/nems/tufuk/RT` on Hercules. You need to check bl_date.conf file to see which date is in there and then look for consistent develop-*

SmithJos13 commented:
Okay it looks like my bl_data.conf  points to  `develop-20250122` but there is no baseline for the test that we want to run `cpld_control_p8`. Unless I'm missing something... 

Here's some info:

```
ls /work2/noaa/nems/tufuk/RT/NEMSfv3gfs/develop-20250122/
```

```
coastal_ike_shinnecock_atm2adc_intel/     coastal_ike_shinnecock_atm2sch_intel/     coastal_irene_atm2roms_gnu/
coastal_ike_shinnecock_atm2sch2ww3_gnu/   coastal_ike_shinnecock_atm2ww3_intel/     coastal_irene_atm2roms_intel/
coastal_ike_shinnecock_atm2sch2ww3_intel/ coastal_ike_shinnecock_ww3_intel/         coastal_scituateharbor_atm2fvc_intel/
```

So you might need to create a new baseline using `feature/coastal_app` branch before we can proceed?

uturuncoglu commented:
@SmithJos13 Since that test belongs to UFS WM, the beeline is not in there. We are just maintaining our tests in that directory. You could run the test (under rt.conf) without modifying directory in the rt.sh.

SmithJos13 commented:
Okay I reverted my rt.sh to what is under `feature/coastal_app` ([See the following link](https://github.com/oceanmodeling/ufs-weather-model/blob/feature/coastal_app/tests/bl_date.conf)) then looking under the directory it lists for the base line we have:

```
ls /work/noaa/epic/hercules/UFS-WM_RT/NEMSfv3gfs/
```
we have 

```
BM_IC-20220207/                     develop-20250612/                   input-data-20240501/
adjust_permission.sh                develop-20250623/                   input-data-20240501_backup/
adjust_permissions.sh               develop-20250630/                   input-data-20240501_backup20241007/
develop-20250502/                   develop-20250701/                   input-data-20250507/
develop-20250528/                   develop-20250717/                   save/
develop-20250603/                   develop-20250721/                   temp/
develop-20250609/                   input-data-20221101/
```

But none of those match my BL as well? do I need update something on my end? 

Running: 
```
./rt.sh -k -l rt.conf -n "cpld_control_p8 intel" -a vdatum
```

yields the following failure:

```
******Regression Testing Script Started******
hercules-login-1.hpc.msstate.edu
Machine: hercules
Account: vdatum
rt.sh: Setting up hercules...
Linking /work2/noaa/vdatum/jsmith/jsmith/stmp/jsmith/FV3_RT/rt_2761929 to /work2/noaa/vdatum/jsmith/Coastal_PR/ufs-weather-model/tests/run_dir
Run regression test in: /work2/noaa/vdatum/jsmith/jsmith/stmp/jsmith/FV3_RT/rt_2761929
rt.sh: Checking & Updating test configuration...
find: ‘/work/noaa/epic/hercules/UFS-WM_RT/NEMSfv3gfs/develop-20250122/’: No such file or directory
```

uturuncoglu commented:
@SmithJos13 It is failing because they delated old baselines and current UFS Coastal is little bit out dated. So, it is little bit complicated/tricky to test your changes. We need to sync your branch with UFS WM to test it. we need to exactly same model except your changes. BTW, we will see it will pass or not. My intention is to first sync UFS Coastal with UFS WM. Then you need to sync your branch respect to UFS Coastal (we need to do it in any case, otherwise we could not know exactly if there is an answer change). The short solution is to create a new beeline from UFS Coastal for that test and then you could compare your changes agains it (I could create develop-20250122 directory with baseline of cpld_control_p8). I'll update you when the baseline is ready. Then, you could run the test to see what happens.

SmithJos13 commented:
@uturuncoglu Sounds good, let me know when that is available, and we can at least have the piece of mind that answer wont change that much when you sync coastal with UFSWM. Then we can figure out how we want to go with integration.

uturuncoglu commented:
@SmithJos13 Sounds great. I am working on it. Let you know soon.

uturuncoglu commented:
@SmithJos13 Okay. I did following,

- I run `cpld_control_p8` test using `feature/sync` branch. Since this is newer version of the UFS Coastal, we have baseline for `cpld_control_p8`. The test passed as I expected since there is no any changes in the UFS Coastal that would affect UFS WM RTs.

- Since I know that current version of UFS Coastal is fine (based on my test with `feature/sync` - it is just updated version of UFS Coastal). I go further and create a baseline for `cpld_control_p8` and i placed the baseline directory to `/work2/noaa/nems/tufuk/RT/NEMSfv3gfs` since I have no write access to `/work/noaa/epic/hercules/UFS-WM_RT/NEMSfv3gfs/develop-20250122`.

So, you could run `cpld_control_p8` test with your branch but do not forget to point `/work2/noaa/nems/tufuk/RT` in your rt.sh. Let me know how it goes.

SmithJos13 commented:
@uturuncoglu SUCCESS!!!!!!!!

```
baseline dir = /work2/noaa/nems/tufuk/RT/NEMSfv3gfs/develop-20250122/cpld_control_p8_intel
working dir  = /work2/noaa/vdatum/jsmith/jsmith/stmp/jsmith/FV3_RT/rt_3851964/cpld_control_p8_intel
Checking test cpld_control_p8_intel results ....
 Comparing sfcf021.tile1.nc .....USING NCCMP......OK
 Comparing sfcf021.tile2.nc .....USING NCCMP......OK
 Comparing sfcf021.tile3.nc .....USING NCCMP......OK
 Comparing sfcf021.tile4.nc .....USING NCCMP......OK
 Comparing sfcf021.tile5.nc .....USING NCCMP......OK
 Comparing sfcf021.tile6.nc .....USING NCCMP......OK
 Comparing atmf021.tile1.nc .....USING NCCMP......OK
 Comparing atmf021.tile2.nc .....USING NCCMP......OK
 Comparing atmf021.tile3.nc .....USING NCCMP......OK
 Comparing atmf021.tile4.nc .....USING NCCMP......OK
 Comparing atmf021.tile5.nc .....USING NCCMP......OK
 Comparing atmf021.tile6.nc .....USING NCCMP......OK
 Comparing sfcf024.tile1.nc .....USING NCCMP......OK
 Comparing sfcf024.tile2.nc .....USING NCCMP......OK
 Comparing sfcf024.tile3.nc .....USING NCCMP......OK
 Comparing sfcf024.tile4.nc .....USING NCCMP......OK
 Comparing sfcf024.tile5.nc .....USING NCCMP......OK
 Comparing sfcf024.tile6.nc .....USING NCCMP......OK
 Comparing atmf024.tile1.nc .....USING NCCMP......OK
 Comparing atmf024.tile2.nc .....USING NCCMP......OK
 Comparing atmf024.tile3.nc .....USING NCCMP......OK
 Comparing atmf024.tile4.nc .....USING NCCMP......OK
 Comparing atmf024.tile5.nc .....USING NCCMP......OK
 Comparing atmf024.tile6.nc .....USING NCCMP......OK
 Comparing gocart.inst_aod.20210323_0600z.nc4 .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.coupler.res .....USING CMP......OK
 Comparing RESTART/20210323.060000.fv_core.res.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_core.res.tile1.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_core.res.tile2.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_core.res.tile3.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_core.res.tile4.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_core.res.tile5.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_core.res.tile6.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_srf_wnd.res.tile1.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_srf_wnd.res.tile2.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_srf_wnd.res.tile3.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_srf_wnd.res.tile4.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_srf_wnd.res.tile5.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_srf_wnd.res.tile6.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_tracer.res.tile1.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_tracer.res.tile2.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_tracer.res.tile3.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_tracer.res.tile4.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_tracer.res.tile5.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.fv_tracer.res.tile6.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.phy_data.tile1.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.phy_data.tile2.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.phy_data.tile3.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.phy_data.tile4.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.phy_data.tile5.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.phy_data.tile6.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.sfc_data.tile1.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.sfc_data.tile2.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.sfc_data.tile3.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.sfc_data.tile4.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.sfc_data.tile5.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.sfc_data.tile6.nc .....USING NCCMP......OK
 Comparing RESTART/20210322.090000.MOM.res.nc .....USING NCCMP......OK
 Comparing RESTART/20210323.060000.MOM.res.nc .....USING NCCMP......OK
 Comparing RESTART/iced.2021-03-22-32400.nc .....USING NCCMP......OK
 Comparing RESTART/iced.2021-03-23-21600.nc .....USING NCCMP......OK
 Comparing RESTART/ufs.cpld.cpl.r.2021-03-22-32400.nc .....USING NCCMP......OK
 Comparing RESTART/ufs.cpld.cpl.r.2021-03-23-21600.nc .....USING NCCMP......OK
 Comparing ufs.cpld.ww3.r.2021-03-22-32400.nc .....USING NCCMP......OK
 Comparing ufs.cpld.ww3.r.2021-03-23-21600.nc .....USING NCCMP......OK
 Comparing 20210323.060000.out_pnt.ww3 .....USING CMP......OK
 Comparing 20210323.060000.out_grd.ww3 .....USING CMP......OK

  0: The total amount of wall time                        = 568.499285
  0: The maximum resident set size (KB)                   = 2203360

Test cpld_control_p8_intel PASS
```

uturuncoglu commented:
@SmithJos13 That is perfect. Did you also run your configuration with this version? Does it look good? If so, we could merge this PR. BTW, are you using exiting CDEPS modes for this case. If I remember correctly, you were using ERA5 for atmosphere and did some changes in the ATMMESH. Did you test with ERA5 mode?

I am not sure why we have some changes in SCHSIM side. This is probably for fully coupled configuration right? and nothing related with DATM-DOCN-CICE one. Right?

Also, please let me know your run directory and I could define a new RT for this configuration. That could be a separate PR once we merge this. BTW, it would be nice to have some level of documentation for this configuration (DATM-DOCN-CICE) in the app level. If you don't mind could you compile a simple doc for me. It does not need to be very detailed one. I think one plot that shows the domain, data used for DATM and DOCN and also CICE specific configurations.

SmithJos13 committed:
updated CDEPS pointed

SmithJos13 committed:
updated pointers

SmithJos13 committed:
updated CDEPS point

SmithJos13 commented:
<img width="869" height="541" alt="Screenshot 2025-07-28 at 11 37 34 AM" src="https://github.com/user-attachments/assets/95e972cb-9850-4969-a3ae-0eb50360706b" />

@uturuncoglu Just missing snow from ERA5 cap... added that and pushed the changes. I'm rerunning the rt to see if it is all working now.  I'll let you know shortly what happens.


Then I'm going to test my configuration. First going to run the shinnecock domain since it is super lightweight and I can skip the line essentially on hercules. If that is producing ice then it should be fine to push the changes since we haven't done extensive validation of the Bering sea model yet.

If all looks good I'll update my Bering Sea configuration and then we can try running that after the Shinnecock one just to be sure. If the Shinnecock one is working it is usually a good sign the Bering one will work.

uturuncoglu commented:
Okay. Please add that field to ERA5 mode. Then you can CDEPS to your PR too. I hope everything goes well with the tests. Let me know if you need help.

SmithJos13 commented:
I did here are the results:

```
Run regression test in: /work2/noaa/vdatum/jsmith/jsmith/stmp/jsmith/FV3_RT/rt_161283
rt.sh: Checking & Updating test configuration...
rt.sh: Running compile s2swa_intel
rt.sh: Compile s2swa_intel completed.
144 6
rt.sh: Running test cpld_control_p8_intel using compile s2swa_intel
rt.sh: Run with test cpld_control_p8_intel completed.
rt.sh: Generating Regression Testing Log...

Performing Cleanup...
REGRESSION TEST RESULT: SUCCESS
******Regression Testing Script Completed******
rt.sh finished
rt.sh: Cleaning up...
rt.sh: Exiting.
```

SUCCESS! Looks like it passed!

I'm going to update the configuration to night when I get some free time and then let you know how it all goes.

uturuncoglu commented:
@SmithJos13 I sorry about not to be clear. This is global case (`cpld_control_p8_intel `), I mean your configuration (DATM-DOCN-CICE6). Anyway, let me know if you need anything from my end.

uturuncoglu commented:
@SmithJos13 I just want to check with you about your progress. I am plaining to merge sync branch today or tomorrow. So, if you are able to finalize your work soon. I could wait for you before merging sync branch.

SmithJos13 commented:
@uturuncoglu I think you should go ahead with the merge. I'm finally getting time to work on adjusting the configuration now. I'll let you know if I can get it working in the next hour or so... otherwise I would just go ahead with out me.

SmithJos13 commented:
@uturuncoglu you should just merge, and I'll deal with it on my end. I have found a couple bugs with the open BCs that need to be fixed on my end anyways.

uturuncoglu commented:
@SmithJos13 Okay. I will go ahead with sync branch. I am just waiting for Frontera spack installation. It would be ready in a couple of hours.

uturuncoglu commented:
@SmithJos13 I have just merged sync branch. Please sync your model and components with the feature/coastal_app. So, we could merge it when you finalize your implementation. Let me know if you need help.

SmithJos13 committed:
merged changed with feature/coastalapp

SmithJos13 committed:
updated CMEPS/CICE pointer

SmithJos13 committed:
updated all submodules

SmithJos13 committed:
tests folder

SmithJos13 committed:
PHAM pointer

SmithJos13 committed:
MOM6 Pointer

SmithJos13 committed:
GOCART Pointer

SmithJos13 committed:
GeoGate Pointer

SmithJos13 committed:
CMakeModules Pointer

SmithJos13 committed:
CDEPS pointer

SmithJos13 committed:
updated CMEPS pointer

SmithJos13 committed:
updated CICE

SmithJos13 commented:
@uturuncoglu did I merge everything correctly? I think I have updated the PR to the newest version of coastal.

uturuncoglu commented:
@SmithJos13 It looks good to me. I could check each component more carefully and let you know. In the meantime, if you could double check your changes in the CICE side that would be great. Also, maybe you could run cpld regression test (from UFS WM) again. BTW, is your case running fine with this version of code?

SmithJos13 commented:
@uturuncoglu I'll run the test again and see what happens. I'm working on fixing my standalone CICE configuration at the moment. updating it to use the new data modes. I'm having issue with getting it to run in parrellel. It is giving me a PMPI_Waitall error which I have never seen before when I try...

```
9: Abort(17) on node 9 (rank 9 in comm 0): Fatal error in PMPI_Waitall: Unknown error class
5: Abort(17) on node 5 (rank 5 in comm 0): Fatal error in PMPI_Waitall: Unknown error class
3: Abort(17) on node 3 (rank 3 in comm 0): Fatal error in PMPI_Waitall: Unknown error class
8: Abort(17) on node 8 (rank 8 in comm 0): Fatal error in PMPI_Waitall: Unknown error class
7: Abort(17) on node 7 (rank 7 in comm 0): Fatal error in PMPI_Waitall: Unknown error class
6: Abort(17) on node 6 (rank 6 in comm 0): Fatal error in PMPI_Waitall: Unknown error class
4: Abort(17) on node 4 (rank 4 in comm 0): Fatal error in PMPI_Waitall: Unknown error class

```

uturuncoglu commented:
@SmithJos13 Maybe the issue is in somewhere else. Could you share your run directory? So, I could check.

SmithJos13 commented:
@uturuncoglu here you go `/work2/noaa/vdatum/jsmith/Coastal_PR/Run_Dir/CICE_Standalone` let me know if you need any permissions I missed

Okay after disabling the open BCs it looks like the problem when away. So it looks like the issues might be stemming form the boundary conditions. I need to check to make sure I didn't delete anything I shouldn't have when I portted it over.

uturuncoglu commented:
@SmithJos13 Could you try to add `export FI_MLX_INJECT_LIMIT=0` to your job script. It might solve the issue with your run.

uturuncoglu commented:
@SmithJos13 Okay. It seems you found the source of the issue. Actually there is a trace in the error file, it might help.

SmithJos13 commented:
Fixed it!!! its working now, I'm going to try to get the standalone configuration fully tested tomorrow and then Ill rerun all our tests

SmithJos13 committed:
update cice pointer

SmithJos13 committed:
updated PHAM pointer

SmithJos13 committed:
updated PHAM pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
updated cice pointer

SmithJos13 committed:
updated cice pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 commented:
Excellent news: IT PASSED!!!

```
baseline dir = /work/noaa/epic/hercules/UFS-WM_RT/NEMSfv3gfs/develop-20250721/cpld_control_p8_intel
working dir  = /work2/noaa/vdatum/jsmith//stmp/jsmith/FV3_RT/rt_2093034/cpld_control_p8_intel
Checking test cpld_control_p8_intel results ....
 Comparing sfcf021.tile1.nc .....USING NCCMP......OK
/...
.../
 Comparing 20210323.060000.out_grd.ww3 .....USING CMP......OK

  0: The total amount of wall time                        = 592.145828
  0: The maximum resident set size (KB)                   = 2258220

Test cpld_control_p8_intel PASS
```

SmithJos13 committed:
updated SCHISM-ESMF pointer

SmithJos13 committed:
updated SCHISM-ESMF pointer

SmithJos13 commented:
```
******Regression Testing Script Started******
hercules-login-1.hpc.msstate.edu
rt.sh: Setting up hercules...
Linking /work2/noaa/vdatum/jsmith//stmp/jsmith/FV3_RT/rt_778664 to /work2/noaa/vdatum/jsmith/Coastal_PR/coastal/tests/run_dir
Run regression test in: /work2/noaa/vdatum/jsmith//stmp/jsmith/FV3_RT/rt_778664
rt.sh: Checking & Updating test configuration...
rt.sh: Running compile atm2sch_intel
rt.sh: Compile atm2sch_intel completed.
3 0
rt.sh: Running test coastal_ike_shinnecock_atm2sch_intel using compile atm2sch_intel
rt.sh: Run with test coastal_ike_shinnecock_atm2sch_intel completed.
rt.sh: Generating Regression Testing Log...

Performing Cleanup...
REGRESSION TEST RESULT: SUCCESS
******Regression Testing Script Completed******
rt.sh finished
rt.sh: Cleaning up...
rt.sh: Exiting.
```

Success! Just waiting for standalone model for the RT.

uturuncoglu commented:
@SmithJos13 That is great news. Let me know when you test your realistic datm-docn-cice configuration and then we could setup a call to merge all the staff.

SmithJos13 commented:
@uturuncoglu looks like Hercules is down today for maintenance so I probably won't be able to verify my model run until tomorrow. or later this evening. Want to work on merging everything on Friday? If it all goes well.

SmithJos13 commented:
@uturuncoglu let circle around to this early next week!

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
updated CMEPS pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 commented:
@uturuncoglu  Things to discuss for today:

I had to make a change to icepack_therm_vertical.f90 to keep the model stable. They had some really bad error checking that was causing a divide by zero error underspecific cases. I made the change and I reran the cpld_control_p8 test and it passed:

```
baseline dir = /work/noaa/epic/hercules/UFS-WM_RT/NEMSfv3gfs/develop-20250721/cpld_control_p8_intel
working dir  = /work2/noaa/vdatum/jsmith/stmp/jsmith/stmp/jsmith/FV3_RT/rt_454435/cpld_control_p8_intel
Checking test cpld_control_p8_intel results ....
Test cpld_control_p8_intel PASS
```
So we are good there. There were no changes to SCHISM so I didn't rerun the SCHISM RT. 


There is one more thing I want to discuss. (We can do this at our meeting today). 

I have been using a linear salinity approximation for the freezing point of the sea water. In this approximation there is a parameter called depressT is a constant that relates salinity to the freezing point. by defualt icepack has this set to 0.054 (degC/ppt) however the ocean state I have been using from ROMS was generated with a simulation that used 0.0543 (degC/ppt). This does make a meaniful difference between the results:

This is a snapshot of the model with **depressT = 0.054**:

<img width="711" height="709" alt="Screenshot 2025-09-05 at 10 38 34 AM" src="https://github.com/user-attachments/assets/18279fc0-960b-4584-af8a-94213c37b3f9" />

and here is one with **depressT = 0.0543**: 

<img width="771" height="781" alt="Screenshot 2025-09-05 at 10 38 25 AM" src="https://github.com/user-attachments/assets/e10b3b95-60b0-431a-af03-582c8e47ccf9" />

In the top panel in each plot is ice thickness. The bottom show shelf average ice area concentration (Bue = model, red = ASI sat). The only thing that changed between these simulations is the value of depressT. You can see there is a meaningful difference. I was wondering how we should proceed with this? I know that this is just a technical integration, but I also wanted to check to see what you wanted for the RT.

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
updated SCHISM pointer

SmithJos13 committed:
updated CMEPS pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
updated CICE pointer

SmithJos13 committed:
commit updated CICE pointer

uturuncoglu commented:
@SmithJos13 I started to merge process and when I cloning your model branch, I am getting following error.

```
Submodule path 'CICE-interface/CICE/icepack': checked out '6f9797a749ef2dbe150a569ccaf151917303b7ad'
fatal: remote error: upload-pack: not our ref bcf1f25738c4aa8cf98519cab6fe8ce1a728c070
Fetched in submodule path 'CMEPS-interface/CMEPS', but it did not contain bcf1f25738c4aa8cf98519cab6fe8ce1a728c070. Direct fetching of that commit failed.
```

So, there might be issue with the submodules. If you don't mind, could you double check? Here is the command that I am using,

```
git clone --recursive -b feature/cice https://github.com/SmithJos13/ufs-weather-model.git ufs-weather-model_cice
```

uturuncoglu commented:
You also need to sync the model. It seems that you have some conflicts.

SmithJos13 commented:
@uturuncoglu I synced the CMEPS model component. I think the clone should work now. Is there anything else that needs to be synced?

uturuncoglu commented:
@SmithJos13 Okay. I could checkout without any issue. It seems that you need to sync model (like .gitmodules) and also SCHSIM-ESMF.

SmithJos13 committed:
updated .gitmodules

SmithJos13 commented:
@uturuncoglu I see. okay glad the checkout has been fixed. How do I resolve this problem? do I have to merge the with the ocean modeling version of feature/coastal_app?

SmithJos13 committed:
updated .gitmodules

uturuncoglu commented:
@SmithJos13 Thanks. Yes. First you need to sync SCHSIM-ESMF in the sub-component level and then you could sync the model with feature/coastal_app branch.

uturuncoglu commented:
@SmithJos13 I think it looks fine for now. Nothing needs to be changed.

SmithJos13 commented:
Okay cool! lmk know if you encounter any more problems

uturuncoglu commented:
@SmithJos13 I am still getting following error from `cpld_control_p8`,

```
20250918 133456.104 INFO             PET175 (ice_import_export:ice_advertise_fields) called
20250918 133456.113 ERROR            PET175 src/addon/NUOPC/src/NUOPC_Base.F90:2128 Invalid argument  - Si_uvel is not a StandardName in the NUOPC_FieldDictionary!
20250918 133456.113 ERROR            PET175 src/addon/NUOPC/src/NUOPC_Base.F90:500 Invalid argument  - Passing error in return code
20250918 133456.113 ER
```

Maybe you forgot to push something to field dictionary?

uturuncoglu commented:
@SmithJos13 I could see that it is added in PR but I could not see `Si_uvel` variable in the fd_ufs.yaml file when I clone the code. Maybe something is wrong. Could you double check?

uturuncoglu commented:
@SmithJos13 If you don't mind, Could you revert back changes in `tests/rt.sh`. We do not want to change the default behavior.

uturuncoglu commented:
BTW, one I run git pull. It get the updated field dictionary. I am not sure why it did not get in the first place. Maybe the branch was pointing the wrong hash.

SmithJos13 commented:
@uturuncoglu Do I need update a hash for fd_yaml? I'll change back rt.sh

SmithJos13 committed:
updated tests/rt.sh and test/parm/fd_ufs.yaml

SmithJos13 commented:
@uturuncoglu okay I tried to fix those. let me know if it works now.

uturuncoglu commented:
@SmithJos13 Starting to merge the development to UFS Coastal. Once I merge sub components, you need to update .gitmodules file and point original repositories. Then, I could merge the model itself.

SmithJos13 committed:
updated CICE pointer

uturuncoglu commented:
@SmithJos13 I merged all sub components. If you don't mind, could you update pointers for CICE, CMEPS, SCHISM and SCHISM-ESMF?

uturuncoglu commented:
@SmithJos13 CDEPS as well. I also merged that one. It is not listed in the description but I'll add it.

Commit:
updated all pointers to feature/coastal_app

Commit:
updated .gitmodules

Commit:
updated all pointers to feature/coastal_app

uturuncoglu commented:
@SmithJos13 You need to point feature/coastalapp for CICE, CMEPS and CDEPS but feature/coastal_app for SCHISM and SCHSIM_ESMF. Sorry for confusion but we do not have convention for the sub components yet in terms of naming. I'll fix them after this PR.

Commit:
updated .gitmodule

Commit:
restored folders

Commit:
reverted folder

Commit:
updated files

Commit:
updated test folder

Commit:
updated test folder

Commit:
updated tests folder

Commit:
updated tests folder

Commit:
updated tests folder

Commit:
updated test/parm folder

Commit:
updated test/parm folder

SmithJos13 commented:
Okay I think it should be good to go

uturuncoglu commented:
@SmithJos13 Thanks. I was expecting to see at least some changes related with the fd_ufs.yaml but I could not see it in the model level. Could you double check it?

Commit:
updated fd_ufs.yaml

SmithJos13 commented:
Okay i reverted the file back to commit cc334d8a25be74ce56009258922e052e2c0f80ba and it looks like I had removed some of the changes. Should be good now.

uturuncoglu commented:
@SmithJos13 Okay. Thanks. We are almost there. Let me do one last minute testing with this config and merge this. Probably, I could merge it today and start working on RT. This is the right folder - /work2/noaa/vdatum/jsmith/Coastal_PR/Run_Dir/CICE_Standalone - for DATM-DOCN-CICE configuration right?

SmithJos13 commented:
@uturuncoglu that is correct location!

NickSzapiro-NOAA commented:
Hi @SmithJos13 @uturuncoglu . There is a regional CICE working group that would like to connect with the regional efforts related to this PR and https://doi.org/10.1029/2025JC022684 . Would y'all be interested in presenting at that regional working group?
